src/core/Akka/Util/AppVersion.cs

```diff
@@ -243,12 +243,12 @@ public int CompareTo(AppVersion other)
 
         public bool Equals(AppVersion other)
         {
-            return other != null && Version == other.Version;
+            return other != null && string.Equals(Version, other.Version, StringComparison.Ordinal);
         }
 
         public override bool Equals(object obj)
         {
-            return base.Equals(obj as AppVersion);
```

Comment:
Since AppVersion extends Object directly, base.Equals() does a direct reference equality, which is wrong.